KeysPerSecond/src/me/roan/kps/Main.java

```diff
@@ -272,31 +279,32 @@ protected static final void mainLoop(){
 		future = timer.scheduleAtFixedRate(new Runnable() {
 			@Override
 			public void run() {
-				if(!suspended){
-					int totaltmp = tmp;
-					for(int i : timepoints){
-						totaltmp += i;
-					}
-					if(totaltmp > max){
-						max = totaltmp;
-					}
-					if(totaltmp != 0){
-						avg = (avg * (double)n + (double)totaltmp) / ((double)n + 1.0D);
-						n++;
-						TotPanel.hits += tmp;
-						System.out.println("Current keys per second: " + totaltmp + " time frame: " + tmp);
-					}
-					graph.addPoint(totaltmp);
-					graph.repaint();
-					content.repaint();
-					prev = totaltmp;
-					timepoints.addFirst(tmp);
-					if(timepoints.size() >= 1000 / config.updateRate){
-						timepoints.removeLast();
-					}
-					tmp = 0;
-				}else{
-					tmp = 0;
+				int currtmp = tmp.getAndSet(0);
+
+				if(suspended) return;
+
+				int totaltmp = timepoints.stream().reduce(currtmp, Integer::sum);

```

Comment:
Since apparently my understanding of autoboxing was slightly off I now assume that a stream for this would be more efficient right? After all the list consists of Integer objects, unboxing them all is probably less efficient then the overhead created by using a stream.